app/src/main/java/com/example/habitsmasher/DaysTracker.java

```diff
@@ -0,0 +1,261 @@
+     */
+    public DaysTracker(){
+        //arraylist of booleans that represent the days of the week. autofilled to 7 bools.
+        _days = new ArrayList<Boolean>(Arrays.asList(new Boolean[7]));
+
+        //fill with all false
+        Collections.fill(_days, Boolean.FALSE);
+    }
+
+
+    /**
+     * Creates a DaysTracker with {@link String} input. The string contains all of the days that are to
+     * be set to true, and all other days false.
+     *<p>
+     * The string takes the format "day1 day2 ... day7", where each day is the first 2 letters of that day, separated by a space. The string can be in any order.
+     * EX) "MO FR WE TH"
+     *
+     * The string is assumed to be formatted correctly, but will not throw any exceptions if not correctly formatted. If any
+     * doubts are present when using this constructor, it may be wise to use the base constructor and set individual days yourself.
+     *
+     * @param days: The string of dates to use, formatted according to above guide.
+     */
+    public DaysTracker(String days){
+
+        _days = new ArrayList<Boolean>(Arrays.asList(new Boolean[7]));
+        Collections.fill(_days, Boolean.FALSE);
+
+        //take the string and split it up based on what days are present
+        //this will go under the assumption that the string will always be formatted correctly.
+        days = days.toUpperCase();
+        String[] splitDays = days.split(" ");
+
+        //for each item split from the string
+        for(int i = 0; i < splitDays.length; i++){
+            //get the value, and if any matches with the abbreviations set the corresponding day to true
+            String value = splitDays[i];
+            switch (value){
+                case "MO":
+                    _days.set(0, true);
+                    break;
+                case "TU":
+                    _days.set(1, true);
+                    break;
+                case "WE":
+                    _days.set(2, true);
+                    break;
+                case "TH":
+                    _days.set(3, true);
+                    break;
+                case "FR":
+                    _days.set(4, true);
+                    break;
+                case "SA":
+                    _days.set(5, true);
+                    break;
+                case "SU":
+                    _days.set(6, true);
+                    break;
+            }
+        }
+
+    }
+
+    /* Was experimenting with hashmaps and iterators, if anyone thinks this may be useful I will complete it,
+     * If not I'll get rid of it.
+
+    public DaysTracker(HashMap<String, Boolean> map){
+        _days = new ArrayList<Boolean>(Arrays.asList(new Boolean[7]));
+        Collections.fill(_days, Boolean.FALSE);
+        Set<String> allKeys = map.keySet();
+
+        //iterator over key set
+        Iterator<String> keyIterator = allKeys.iterator();
+        while(keyIterator.hasNext()){
+            switch(keyIterator.next()){
+
+            }
```

Comment:
When I was creating the constructors, I debated whether to use hashmaps or strings. I settled on strings, but if anyone thinks having a hashmap based constructor is important I'll finish it.

Reply:
I do think implementing this with a hashmap would be neater, but it's really not a **huge** deal if you already have the current implementation working well. You can create a task to refactor the DaysTracker if you really feel like it, but it's not the biggest deal to me personally!

Reply:
I agree that if the current implementation is working fine, we should move on rather than spend extra time on a small change like this. Refactoring can be done later if it proves to be important.

Reply:
In that case, I'll leave the code fragment there, and delete it if we never end up needing it.